octobot_backtesting/data/database.py

```diff
@@ -162,12 +162,12 @@ def __selected_columns(self, columns=None):
                                       f"{additional_clauses} {group_by}")
             return await self.cursor.fetchall() if size == self.DEFAULT_SIZE else await self.cursor.fetchmany(size)
         except OperationalError as e:
-            if not await self.__check_table_exists(table):
+            if not await self.check_table_exists(table):
                 raise DataBaseNotExists(e)
             self.logger.error(f"An error occurred when executing select : {e}")
         return []
 
-    async def __check_table_exists(self, table) -> bool:
+    async def check_table_exists(self, table) -> bool:
```

Comment:
Why not `_check_table_exists` ?

Reply:
Because it's now used in API to check if updaters are to be created

Reply:
I don't think this is the correct way of checking if an updater should be created.

Reply:
I think we need a `is_data_available` method or something like this that checks if an updater will have enough data to work.

Reply:
In this method there will be a :
- `_check_table_exists`
- Maybe a check that this table is not empty
...

Reply:
that's what ```def get_available_data_types(importer) -> list:``` from octobot_backtesting/api/importer.py is doing which in then being used in exchange simulator

Reply:
Ok I agree with this code, but I don't understand why you only check `check_table_exists` in `_init_available_data_types` ? I think you should use an equivalent of `get_available_data_types`

Reply:
You mean also checking that the table isn't empty ?

Reply:
Yes. I mean if we check before creating the producer (it's a good idea), why not properly check if the producer has to be created.

Reply:
Ok, I will add it 👍